doc/lib/mqtt.txt

```diff
@@ -0,0 +1,60 @@
+/**
+@mainpage
+@anchor mqtt
+@brief MQTT 3.1.1 client library.
+
+MQTT is a standardized publish subscribe protocol popular on embedded devices. The MQTT library provided with this SDK implements a subset of the MQTT v3.1.1 protocol. The sample application for Linux `subscribe_publish_sample.c` is provided as a starting point for using this library.
+
+The interface for communication over MQTT is provided in the file `aws_iot_mqtt_interface.h`.
+- MQTT client context management: @ref mqtt_function_init and @ref mqtt_function_free
+- Connection management: @ref mqtt_function_connect and @ref mqtt_function_disconnect
+- Publishing messages to the server: @ref mqtt_function_publish
+- Managing subscriptions: @ref mqtt_function_subscribe and @ref mqtt_function_unsubscribe
+- Process incoming messages, reconnections, and keep-alive: @ref mqtt_function_yield
+
+@note In a multithreaded environment, always acquire a lock before calling any function in this library. This library is not thread safe.
```

Comment:
Can this note be updated to point to the multi-threaded recommendation text in the PortingGuide instead ? The customer could either use mutex-based synchronization or choose to serialize all MQTT calls in a thread-safe queue.  This text seems to be suggestion only the first option.

Reply:
Updated wording